LenovoLegionToolkit.Lib/Structs.cs

```diff
@@ -369,16 +369,36 @@ public enum SourceType
             CapabilityData
         }
 
-        public SourceType Source { get; init; }
-        public bool IGPUMode { get; init; }
-        public bool AIChip { get; init; }
-        public bool FlipToStart { get; init; }
-        public bool NvidiaGPUDynamicDisplaySwitching { get; init; }
-        public bool InstantBootAc { get; init; }
-        public bool InstantBootUsbPowerDelivery { get; init; }
-        public bool AMDSmartShiftMode { get; init; }
-        public bool AMDSkinTemperatureTracking { get; init; }
-        public bool GodModeFnQSwitchable { get; init; }
+        private readonly HashSet<CapabilityID> _capabilities = [];
+
+        public SourceType Source { get; }
+
+        public IEnumerable<CapabilityID> All => _capabilities.Order().AsEnumerable();
+
+        public FeatureData(SourceType sourceType)
+        {
+            Source = sourceType;
+        }
+
+        public FeatureData(SourceType sourceType, IEnumerable<CapabilityID> capabilities)
+        {
+            Source = sourceType;
+
+            foreach (var capability in capabilities)
+                _capabilities.Add(capability);
+        }
+
+        public bool this[CapabilityID key]
+        {
+            get => _capabilities.Contains(key);
+            init
+            {
+                if (value)
+                    _capabilities.Add(key);
+                else
+                    _capabilities.Remove(key);
+            }
+        }
```

Comment:
There are two problems with your suggestions:
1. indexer logic is changed - it will add a capability to the set even if boolean value was false
2. you missed the other constructor which will break the code, because set is never initialized